### DISCUSSION TITLE
C6 / IC4 build

### DISCUSSION THREAD

**Discussion by chinasaur** [doudar/SmartSpin2k#239] [Show and tell]
I finally got around to putting some things together and testing. It's looking okay so far. My build is a bit non-standard, partly because I am trying to specialize things for the C6 / IC4, and partly just due to what parts I was able to find.

For shifting I'm trying to use the built-in up/down buttons on the bike's console. I'm accessing the button presses by intercepting the FFC ribbon cable. I also added a voltage divider, but it may not be needed; it turns out these button lines are already down at 3.7V, which might be okay for the ESP32. Since it's there, I'm using the voltage divider just in case. ![image](https://user-images.githubusercontent.com/68619/128621149-bcfb5957-546c-4f7d-bfcf-0429eeaf6611.png)

For motor I'm trying [this](https://www.pololu.com/product/1208). It's a bit smaller, but seems to have plenty of torque to turn the C6 resistance knob in my tests, even just using the 9V available from the C6's built-in DC adapter. I'm not sure yet if I will try to run it off the 9V in practice, or use a separate higher voltage source for the motor.

For stepper controller I got something like [this](https://wiki.fysetc.com/TMC2225/). It seems to run fine on 3.3V. I haven't tested the UART interface yet, but it has the series resistor already on the board which hopefully simplifies things a bit.

Hopefully I'll have more to report soon. On the software / wiring side it's mostly compatible. But I'm anticipating needing to customize some things, like my up/down are I think pin-reversed and are also idle HIGH. I would also like to use the resistance sensor readings as feedback for the motor control; is that already in the code?

**Comment by chinasaur** [comment]
The carrier board is working okay too:
![image](https://user-images.githubusercontent.com/68619/129945212-95d350be-b38d-4afe-9716-ba999dacecee.png)

This has the usual connections between ESP32, TMC2225, stepper, and shifter lines. It also has headers to intercept the C6/IC4 8-pin harness (physically 10-pin with 2 unused) and some currently unpopulated transistor pads to enable voltage shifted I2C communication with the bike's resistance sensor. This is in case I want to talk to the resistance sensor directly rather than relying on what's available over bluetooth.

For ESP32 power I plan to use the existing USB A charger port on the bike console. For motor power I'm leaning towards using the 9V from the bike harness. The 9V adapter that comes with the bike is rated for 1.5 A, and the motor should pull less than 0.5 A per coil at 9V. So I may do another revision of this carrier to clean things up further.

**Reply by doudar** [reply]
I'd love to see you push the carrier board once you're happy with it. 

Nice work!

Also, you should give the 2.0 case a try - It's all around better and I bet you can find room for your carrier board.

**Comment by tootechy** [comment]
Slight side note, I'm using a Tempo Power meter on my IC4. They modify the crank arm for you: https://www.tempopowermeters.com/product-page/spin-bike-power-meter

Can you comment on how it working w/ your bike? I'm in the process of sourcing the parts since the Amazon links are out of stock.

**Reply by chinasaur** [reply]
Interesting; power crank looks nice to have as I have found the power reported to Zwift by my C6 is pretty far off, and I'm reluctant to recalibrate it. I'll try to adjust the power in software once I have the SmartSpin in between.

In terms of my build, unfortunately I haven't had time to test the new carrier board yet. It's ready to go though; hopefully update soon.
